misk/src/main/kotlin/misk/web/interceptors/EarlyBind.kt

```diff
@@ -0,0 +1,20 @@
+package misk.web.interceptors
+
+import misk.MiskDefault
+import misk.web.NetworkInterceptor
+import javax.inject.Qualifier
+
+/**
+ * Denotes an early binding target, which will be automatically installed.
+ * For example, the [NetworkInterceptor] is bound with [EarlyBind] and is automatically
+ * installed before [MiskDefault] and non-annotated user provided interceptors.
+ */
+@Qualifier
+@Retention(AnnotationRetention.RUNTIME)
+@Target(
+  AnnotationTarget.CLASS,
+  AnnotationTarget.VALUE_PARAMETER,
+  AnnotationTarget.FIELD,
+  AnnotationTarget.TYPE
+)
+annotation class EarlyBind
```

Comment:
Hmmm . . . I was thinking ‘Early’ in the sense that the execute before the other interceptors. The word _bind_ here suggests that they’re bound early?

I’d like another word for this. Maybe something about them being at the beginning of the list? Or executed first? 

Another option – we could make it clear that these ones run before the content is transformed?

`@BeforeContentEncoding` ? That’s very direct and non-surprising.

Reply:
Naming is hard. I'll take `@BeforeContentEncoding` and also introduce another internal annotation `@ContentEncoding` so that the namespaces make sense